torch/ao/quantization/utils.py

```diff
@@ -495,35 +495,26 @@ def forward(self, x, key1=3, key2=3):
     root = model
     fqn_to_example_inputs = {}
 
-    class InterceptionModule(type(model)):  # type: ignore[misc]
-        def __call__(self, *args, **kwargs):
-            orig_module_call = torch.nn.Module.__call__
-
-            def _patched_module_call(self, *args, **kwargs):
-                submodule_example_inputs = list(args).copy()
-                normalized_kwargs = _normalize_kwargs(self.forward, kwargs)
-                # minus 1 to skipping counting `self`
-                num_args = _get_num_pos_args(self.forward) - 1
-                num_to_pop = num_args - len(submodule_example_inputs)
-                while num_to_pop and normalized_kwargs:
-                    normalized_kwargs.popitem(last=False)
-                    num_to_pop -= 1
-                submodule_example_inputs.extend(normalized_kwargs.values())
-                submodule_example_inputs_tuple = tuple(submodule_example_inputs)
-                fqn = _get_path_of_module(root, self)
-                if fqn is not None:
-                    fqn_to_example_inputs[fqn] = submodule_example_inputs_tuple
-                return orig_module_call(self, *args, **kwargs)
-
-            torch.nn.Module.__call__ = _patched_module_call
-            super().__call__(*args, **kwargs)
-            torch.nn.Module.__call__ = orig_module_call
-
-    original_class = model.__class__
-    model.__class__ = InterceptionModule
+    def _patched_module_call(self, *args, **kwargs):
+        submodule_example_inputs = list(args).copy()
+        normalized_kwargs = _normalize_kwargs(self.forward, kwargs)
+        # minus 1 to skipping counting `self`
+        num_args = _get_num_pos_args(self.forward) - 1
+        num_to_pop = num_args - len(submodule_example_inputs)
+        while num_to_pop and normalized_kwargs:
+            normalized_kwargs.popitem(last=False)
+            num_to_pop -= 1
+        submodule_example_inputs.extend(normalized_kwargs.values())
+        submodule_example_inputs_tuple = tuple(submodule_example_inputs)
+        fqn = _get_path_of_module(root, self)
+        if fqn is not None:
+            fqn_to_example_inputs[fqn] = submodule_example_inputs_tuple
+        return orig_module_call(self, *args, **kwargs)
+
+    orig_module_call = torch.nn.Module.__call__
```

Comment:
should this code be in a `try` followed by L517 in a `finally`, to ensure that the call function gets patched back even if the forward encounters an exception?

Reply:
yeah sure